lib/Extractor/Solver.cpp

```diff
@@ -95,6 +99,20 @@ class BaseSolver : public Solver {
         Guesses.emplace_back(IC.getConst(APInt(LHS->Width, -1)));
       for (auto I : Guesses) {
         InstMapping Mapping(LHS, I);
+
+        if (EnableAliveTest) {
+          Inst *Ante = IC.getConst(APInt(1, true));
+          for (auto PC : PCs ) {
+            Inst *Eq = IC.getInst(Inst::Eq, 1, {PC.LHS, PC.RHS});
+            Ante = IC.getInst(Inst::And, 1, {Ante, Eq});
+          }
+          bool IsValid = isTransformationValid(Mapping.LHS, Mapping.RHS, Ante);
+          if (IsValid) {
+            RHS = I;
+          }
+          return std::error_code(); // TODO: Propagate errors from Alive backend
```

Comment:
don't unconditionally return here, this piece only synthesized 0, 1, -1
we still want to synthesize arbitrary integers